iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -524,12 +524,18 @@ static int parse_device_twin_topic_info(const char* resp_topic, bool* patch_msg,
                 else if (token_count == 3)
                 {
                     *status_code = (int)atol(STRING_c_str(token_value));
+                    *patch_msg = false;
                     if (STRING_TOKENIZER_get_next_token(token_handle, token_value, "/?$rid=") == 0)
                     {
                         *request_id = (size_t)atol(STRING_c_str(token_value));
+                        result = 0;
+                    }
+                    else
+                    {
+                        LogError("Could not parse request_id.");
+                        *request_id = 0;
```

Comment:
Hi @fgiancane8 , just one comment, would you mind changing this line of code and verifying again against your compiler?
As a guideline we do not set out variables if the function fails. 
LSS, line 536 should be removed (unless your compiler complains about that).

Reply:
Hi @ewertons ,

Thanks for the feedback. I added the `*request_id = 0` assignment because the esp32 compiler (which is a gcc with `-Wall` `-Wextra` flags) complained about usage of a not initialised variable (`-Wmaybe-uninitialized` flag).

There are no real cases of not having the request id, unless serious bugs in place.

But if, as guideline, you do not set the return values when function fails, I think I can rework the PR to explicitly initialise `request_id` in the caller fixing the issue.

Reply:
I also take the opportunity to point out that #1394 (which is another PR of mine) addresses same issues (errors found by enabling `-Wall -Wextra`. Could you please have a look at them?

Reply:
Hi @ewertons ,

I updated the PR in order to be compliant with guidelines.

However I am not completely happy of the result as `request_id` is initialized to a valid request ID from Azure point of view.

What do you think if we change the `request_id` type from `size_t` to `ssize_t` (or any other signed integer alias) so that we can explicitly initialize request_id to -1 which is invalid from the Cloud point of view.

This is quite a philosophical reasoning, as if the function `parse_device_twin_topic_info` fails the branch taking into account request_id is not executed.
Changing the variable type is not also an interface breaking as it is used in just the function `mqtt_notification_callback` which is static.

Thanks,
Francesco

Reply:
LSS, 0 for request_id for the failing branch is OK, -1 would be cleaner.